tools/pinning/oldest/pyproject.toml

```diff
@@ -67,7 +67,7 @@ ndg-httpsclient = "0.3.2"
 parsedatetime = "2.4"
 pbr = "1.8.0"
 ply = "3.4"
-pyOpenSSL = "17.5.0"
+pyOpenSSL = "24.1.0"
```

Comment:
What new functionality of pyOpenSSL/cryptography are we using that requires us to bump this? I thought the point of this PR was that things were being deprecated in newer versions. Will we no longer work with the old versions after this change?